### DISCUSSION TITLE
News announcements related to this repo

### DISCUSSION THREAD

**Discussion by lcn2** [ioccc-src/winner#18] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by xexyl** [comment]
> ## 👋 Welcome!
> We’re using Discussions as a place to connect with other members of our community. We hope that you:
> 
> * Ask questions you’re wondering about.

Well in that case! :-)

I'm not clear on what to do with the markdown when it comes to the footer you have at the end of the markdown files. In a recent pull request (today) I just manually edited the html file as well to save the effort of having to edit the footer in (like I did last time) as there weren't many textual changes.

> * Share ideas.

I didn't want to do it since it doesn't concern only my entries but...

```sh
$ git status|grep -oc '.DS_Store'  
15
```

In other words you might want to put at the root directory `.gitignore` the has the glob to prevent this.

Another thing: you might (not sure how this will go and it might be a problem with some old emails no longer existing so a lot of bounces) want to email past winners to let them know about the repo and maybe have them watch this repo. I only started watching this repo today when a comment suggested that winners probably will be using GitHub (I only started using it because I wanted to contribute to mkiocccentry - and it ended up being much more than that of course) already (I had no idea how to set it up or anything! I know how to use git but did not know how to use GitHub). That same comment suggested that they might be watching this repo. But I forgot to do that.

If email will not work maybe tweets will? Or maybe this is not something to pursue: but it's still something to remember that not everyone will be using GitHub (maybe you should put in the future guidelines that it's highly recommended to set up a GitHub account? See below also on this) but if it concerns their entry it will be relevant.

As the comment in that thread started:

> Currently, question 4 of [the FAQ](https://www.ioccc.org/faq.html) states:
> 
> > Q: I found a bug in a previous winner, what should I do?
> > A: We do not ``maintain'' the contest winners. The code is made available on an ``AS IS'' basis. If you have a FIX for a entry, we suggest that you try and EMail it to the authors and to CC: us using the instructions on the contacting the IOCCC Judges page.
> > IF you have a FIX, and IF the author does not object, and IF your fix looks reasonable, we will add a web link to a copy of your EMail message for others to see.
> 
> Would it make sense to direct corrections to the issue tracker and/or the Pull Requests section? Chances are that _if_ the original author is still active, they are likely on GitHub and perhaps already watching the IOCCC winner repository. Even if not, kindly asking the contributor to file a pull request in addition to emailing the author and judges could save some time for the author as well as the judges.
> 
> Moreover, would it make sense to mention the issue tracker in [the contact page](https://www.ioccc.org/contact.html) as well, e.g. for public questions and/or correction proposals to the IOCCC site itself?

Personally I would like to have (and I think other winners would probably too) a say in others editing my winning entries and for some things there are very specific things that must remain the same (like the Enigma machine enciphered files as a big example!).

As for GitHub this page might be useful to some: [Setup SSH keys for use with GitHub/GitLab/BitBucket etc](https://gist.github.com/developius/c81f021eb5c5916013dc).

> * Engage with other community members.
> * Welcome others and are open-minded. Remember that this is a community we
>   build together 💪.
> 
> To get started, comment below with an introduction of yourself and tell us about what you do with this community.

I don't even know what to say with this one at least if I'm to remain humble. Perhaps I'm the one who dedicated a certain tool dealing with tarballs to those who have been tarred and feathered? :-)

> "Because sometimes people throw feathers on tar." :-(

There's more I can say but I have to run! I've started to watch this repo now but hopefully the comments above will be of use to get previous winners here in case they're not - plus the other things.

Good day my friend! Hope the power situation in Canada is doing okay! BCNU! :-)

**Comment by xexyl** [comment]
Oh btw I wanted to remind you that Yusuke Endoh did a write-up of the past winners where he includes ways to get them compiled too if there were problems. It's in Japanese but you can use Google Translate. He did tell me that there are some unfortunate things with Japanese -> English so for example where it looks like he said I wrote 'ridiculously long' for Snake it just meant it's a lot of text (which is true - and some might say some of it is unnecessary but I like to be thorough .. at least when it comes to the IOCCC):

https://mame-github-io.translate.goog/ioccc-ja-spoilers/?_x_tr_sl=auto&_x_tr_tl=en&_x_tr_hl=en-US&_x_tr_pto=wapp

**Comment by lcn2** [comment]
# NOTICE OF UNAVAILABILITY

Starting today, 2024 Oct 29, and increasing over the next several days, we will be less and less active online including in this GitHub repository. This is because we will in a few days, we will be working in an area of the world where there is little to no reliable Internet services and/or the Internet services will be too slow and power too limited to do much work.

There may be rare occasions where we do have access to online services, but such access will be limited to only essential activities. We might be able to even push through an update, but please don't count on it.

And we certainly won't have the opportunity to read messages and respond while we are "remote". So if you write something and we don't respond, you will hopefully remember this notice and understand.

We expect to start to return to normal Internet services by mid November and the goal of returning to full Internet service by 2024 Nov 21.

Best wishes 👋🏻.

**Reply by xexyl** [reply]
Best wishes to you too! Stay safe and well and BCNU!

**Comment by lcn2** [comment]
Just as an FYI:

The next update to this repo, and it will be a big one, will be as a result of the **Great Fork Merge**.

For information on the **Great Fork Merge** and our progress, please see [temp-test-ioccc issue 2239](https://github.com/ioccc-src/temp-test-ioccc/issues/2239) and especially the [top comment of temp-test-ioccc issue 2239](https://github.com/ioccc-src/temp-test-ioccc/issues/2239#issue-2170848675).  You can track the items left TODO (unchecked [ ] boxes).

**Comment by lcn2** [comment]
We are working hard to meet a code freeze by 2024 Feb 28, followed by final testing and a rules/guidelines/FAQ editing review in time for the 2025 March 5 opening of the 40th anniversary IOCCC28. So far so good 👍

**Reply by xexyl** [reply]
I think I just figured out a solution to another problem I discovered in chkentry. I'm working on testing it now. If so I should have no problem doing the `-i` option in time. But of course since the contest does not technically need it (as `mkiocccentry` will not use that option) it could I guess wait until after the contest. Still I hope that this will work. If it does I'll include it in the commit I will make later today (at least the set up).

I have used this technique before so I guess it should work.
